administrator/components/com_installer/Helper/InstallerHelper.php

```diff
@@ -157,4 +162,87 @@ public static function getStateOptions()
 
 		return $options;
 	}
+
+	/**
+	 * Get the Installation XML of a given extension
+	 *
+	 * @param   object  $extension  item from #__extensions
+	 *
+	 * @return  SimpleXMLElement  Installation XML
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public static function getInstallationXML($extension)
+	{
+		if ($extension->client_id)
+		{
+			$path = JPATH_ADMINISTRATOR;
+		}
+		else
+		{
+			$path = JPATH_ROOT;
+		}
+
+		switch ($extension->type)
+		{
+			case 'component':
+				$path .= '/components/' . $extension->element . '/' . substr($extension->element, 4) . '.xml';
+				break;
+			case 'plugin':
+				$path .= '/plugins/' . $extension->folder . '/' . $extension->element . '/' . $extension->element . '.xml';
+				break;
+			case 'module':
+				$path .= '/modules/' . $extension->element . '/' . $extension->element . '.xml';
+				break;
+			case 'template':
+				$path .= '/templates/' . $extension->element . '/templateDetails.xml';
+				break;
+			case 'library':
+				$path = JPATH_ADMINISTRATOR . '/manifests/libraries/' . $extension->element . '.xml';
+				break;
+			case 'file':
+				$path = JPATH_ADMINISTRATOR . '/manifests/files/' . $extension->element . '.xml';
+				break;
+			case 'package':
+				$path = JPATH_ADMINISTRATOR . '/manifests/packages/' . $extension->element . '.xml';
+		}
+
+		return $installXmlFile = simplexml_load_file($path);
+	}
+
+	/**
+	 * Get the download key of an extension going through their installation xml
+	 *
+	 * @param   object  $extension  element of an extension
+	 *
+	 * @return  array  An array with the prefix, suffix and value of the download key
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public static function getDownloadKey($extension)
+	{
+		$installXmlFile = getInstallationXML($extension);
```

Comment:
This won't work because getInstallationXML() is not a regular function but part of this class. Now it generates a fatal error. So you need to call it as ```$installXmlFile = self::getInstallationXML($extension);```